lib/administration/resource/include/irods/resource_administration.hpp

```diff
@@ -176,139 +307,110 @@ namespace irods::experimental::administration
+            input.arg2 = _name.data();
+
+            if constexpr (std::is_same_v<Property, resource_type_property>) {
+                input.arg3 = "type";
+                input.arg4 = _property.value.data();
+            }
+            else if constexpr (std::is_same_v<Property, host_name_property>) {
+                input.arg3 = "host";
+                input.arg4 = _property.value.data();
+            }
+            else if constexpr (std::is_same_v<Property, vault_path_property>) {
+                input.arg3 = "path";
+                input.arg4 = _property.value.data();
+            }
+            else if constexpr (std::is_same_v<Property, resource_status_property>) {
+                input.arg3 = "status";
+
+                if (_property.value == resource_status::up) {
+                    input.arg4 = "up";
+                }
+                else if (_property.value == resource_status::down) {
+                    input.arg4 = "down";
+                }
+                else {
+                    THROW(SYS_INVALID_INPUT_PARAM, "Invalid value for resource status");
+                }
+            }
+            else if constexpr (std::is_same_v<Property, resource_comments_property>) {
+                input.arg3 = "comment";
+                input.arg4 = _property.value.data();
+            }
+            else if constexpr (std::is_same_v<Property, resource_info_property>) {
+                input.arg3 = "info";
+                input.arg4 = _property.value.data();
+            }
+            else if constexpr (std::is_same_v<Property, free_space_property>) {
+                input.arg3 = "free_space";
+                input.arg4 = _property.value.data();
+            }
+            else if constexpr (std::is_same_v<Property, context_string_property>) {
+                input.arg3 = "context";
+                input.arg4 = _property.value.data();
+            }
+
+            if (const auto ec = rxGeneralAdmin(&_comm, &input); ec < 0) {
+                constexpr char* msg = "Could not change {} from [{}] to [{}] for resource [{}]";
+                // NOLINTNEXTLINE(cppcoreguidelines-pro-bounds-array-to-pointer-decay)
+                THROW(ec, fmt::format(msg, to_string<Property>(), input.arg3, input.arg4, _name));
+            }
+        } // modify_resource
+
+        /// Starts a rebalance on a resource.
         ///
         /// \param[in] _comm The communication object.
         /// \param[in] _name The name of the resource.
         ///
         /// \returns A std::error_code.
-        auto rebalance_resource(RxComm& _comm, const resource_name_type _name) -> std::error_code;
+        auto rebalance_resource(RxComm& _comm, const resource_name_type _name) -> void;
 
-        /// \brief Retrieves the name of a resource given a resource ID.
+        /// Retrieves the name of a resource given a resource ID.
         ///
         /// \param[in] _comm The communication object.
         /// \param[in] _name The ID of the resource.
```

Comment:
```suggestion
        /// \param[in] _id The ID of the resource.
```
Needs some `\throws irods::exception` as well, I think